uwblueprint/planet-read
# Add authorization checks for editing translations

## Notion ticket link

[Add authorization checks for editing translations](https://www.notion.so/uwblueprintexecs/Add-authorization-checks-for-editing-translations-3d86e674e9984437ae4560d3eb6ae669)



## Implementation description
* updated params to include user_id
* checked if the user_id was the same as the translator_id for the story translation 
* **note**: this is a temporary change (as discussed with @gaoxk) - ideally, the user_id should not be passed, but there is a bug with the authToken not being passed in the request header


## Steps to test
1. Log in through the frontend
2. Go to one of your story translations, or assign yourself one
3. Edit the story translation contents (this should work)
4. To check an unauthorized translation, try this [query](http://localhost:5000/graphql?variables=null&operationName=updateStoryTranslations&query=mutation%20updateStoryTranslations%20%7B%0A%20%20updateStoryTranslationContentById(storyTranslationContentData%3A%20%7BtranslationContent%3A%20%22hello%22%2C%20id%3A%2011%7D%2C%20userId%3A%201)%20%7B%0A%20%20%20%20story%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%20%20lineIndex%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A) and update the params so that the storyTranslationContent belongs to a story translation the user is not assigned (this should not work). 



## What should reviewers focus on?
* query to obtain translator_id

## Checklist
